Source/VMware.vSphereDSC/DSCResources/VMHostCache.ps1

```diff
@@ -109,11 +112,19 @@ class VMHostSSDCache : VMHostBaseDSC {
     <#
     .DESCRIPTION
 
-    Converts the passed MB value to GB value by rounding it down.
+    Converts the passed MB value to GB value by rounding it down with 3 fractional digits in the return value.
+    #>
+    [double] ConvertMBValueToGBValue($mbValue) {
+        return [Math]::Round($mbValue * $this.MegabytesInGB * $this.MegabytesInGB / 1GB, $this.NumberOfFractionalDigits)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Converts the passed GB value to MB value by rounding it down.
     #>
-    [int] ConvertMBValueToGBValue($mbValue) {
-        $megabytesInGB = 1024
-        return [int] [Math]::Floor($mbValue * $megabytesInGB * $megabytesInGB / 1GB)
+    [long] ConvertGBValueToMBValue($gbValue) {
+        return [long] [Math]::Floor($gbValue * $this.MegabytesInGB * $this.MegabytesInGB * $this.MegabytesInGB / 1MB)
```

Comment:
Conversion should be

$gbValue*1GB/1MB

Reply:
Done.